NERODevelopment/src/controllers/offviewcontroller.cpp

```diff
@@ -0,0 +1,50 @@
+#include "offviewcontroller.h"
+
+OffViewController::OffViewController(Model *model, QObject *parent)
+    : QObject{parent}, m_packTemp(0.0), m_motorTemp(0.0), m_stateOfCharge(0.0) {
+    this->m_model = model;
+    connect(m_model, &Model::updateAttributesFromModel, this,
+            &OffViewController::updateAttributesFromModel);
+}
+
+QMap<QString, AttributeStatus> OffViewController::getAttributeStatus() const { return m_attributeStatus; }
+void OffViewController::setAttributeStatus(const QString &name, AttributeStatus status)
+{
+    if (m_attributeStatus.value(name) != status)
+    {
+        m_attributeStatus[name] = status;
+        emit attributeStatusChanged(m_attributeStatus);
+    }
+}
+
+void OffViewController::updateAttributesFromModel() {
+    const auto attributeStatusMap = m_model->getAttributeStatus();
+    for (const auto& attributeName : {SIDEBRBS, BMS, BSPD, MPU, BOTS, INERTIA, CPBRB, TSMS, HVDINTRLK, HVCNCTR}) {
```

Comment:
you also need to update the pack temp, motor temp and state of charge, the model already has dedicated functions for these